test/test.js

```diff
@@ -231,6 +231,75 @@ describe('failure tests', () => {
       .end(done);
   });
 
+
+  it('should throw if deffered provider throws', done => {
```

Comment:
I think when I suggested this test, I envisaged testing an error thrown directly, rather than wrapped in a promise. But Ive now remembered that since the location where this is called uses `await`, both a thrown error and a rejected promise are handled the same and converted into a thrown error. So, this first test (lines 235 to 255) can be removed as it is functionally equivalent to the one below it.